### ISSUE TITLE
Internal compiler error: mutex.rs could not find scope info

### ISSUE THREAD

**Issue by cbcrouch** [rust-lang/rust#23053]
Received the following error while compiling a test application in rust. If this looks like a geuine error I can provide source, stack traces, etc. if needed.

```
/home/uuser/Tools/rust/rust_dev/src/libstd/sync/mutex.rs:177:37: 180:2 error: internal compiler error: debuginfo: Could not find scope info for node NodeExpr(Expr { id: 4974, node: ExprStruct(Path { span: Span { lo: BytePos(4553404), hi: BytePos(4553415), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: StaticMutex#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }, [Field { ident: Spanned { node: lock#0, span: Span { lo: BytePos(1770102), hi: BytePos(1770106), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 4975, node: ExprPath(None, Path { span: Span { lo: BytePos(4553428), hi: BytePos(4553443), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: sys#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: MUTEX_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4553428), hi: BytePos(4553443), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4553422), hi: BytePos(4553443), expn_id: ExpnId(4294967295) } }, Field { ident: Spanned { node: poison#0, span: Span { lo: BytePos(1770129), hi: BytePos(1770135), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 4976, node: ExprPath(None, Path { span: Span { lo: BytePos(4553457), hi: BytePos(4553474), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: poison#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: FLAG_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4553457), hi: BytePos(4553474), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4553449), hi: BytePos(4553474), expn_id: ExpnId(4294967295) } }], None), span: Span { lo: BytePos(4553404), hi: BytePos(4553477), expn_id: ExpnId(4294967295) } })
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', /home/uuser/Tools/rust/rust_dev/src/libsyntax/diagnostic.rs:129
```

**Comment by cbcrouch** [comment]
Forgot to attach this.

stack backtrace:
   1:     0x7faa4ec31c1f - sys::backtrace::write::h4a039d759790405fOBA
   2:     0x7faa4ec5c932 - panicking::on_panic::h38cda3cfda05a4d8hHJ
   3:     0x7faa4eb91bda - rt::unwind::begin_unwind_inner::hbc6ba0bf039eeb6dknJ
   4:     0x7faa4bfc708d - rt::unwind::begin_unwind::h13059210073216592212
   5:     0x7faa4bfc7033 - diagnostic::SpanHandler::span_bug::h531e8213fd4b4588h0D
   6:     0x7faa4c8a2cf3 - session::Session::span_bug::h12f8cbfb39f8a605ISp
   7:     0x7faa4e3badfc - trans::debuginfo::scope_metadata::h81ce967f070c060bSjE
   8:     0x7faa4e2cd9f8 - trans::debuginfo::set_source_location::h5065a2bc414e2ce68MD
   9:     0x7faa4e282432 - trans::expr::trans_into::h77ab0e7120a822e9znh
  10:     0x7faa4e2826f9 - trans::expr::trans_into::h77ab0e7120a822e9znh
  11:     0x7faa4e2e4c6f - trans::expr::trans_uniq_expr::hb45ab4b4566d896cukj
  12:     0x7faa4e2e58df - trans::expr::trans_unary::habacd13d7a8861e7Jgj
  13:     0x7faa4e2cfdfe - trans::expr::trans_unadjusted::h99b640dbc3358c2cz4h
  14:     0x7faa4e282a07 - trans::expr::trans_into::h77ab0e7120a822e9znh
  15:     0x7faa4e302ce9 - trans::expr::trans_adt::hc28a4aed96db1eb6O6i
  16:     0x7faa4e30579f - trans::expr::trans_struct::closure.42069
  17:     0x7faa4e2ef606 - trans::expr::trans_struct::h90194012c12d3616K2i
  18:     0x7faa4e2d1b2d - trans::expr::trans_rvalue_dps_unadjusted::hba1263ec7559b76b4zi
  19:     0x7faa4e2829e6 - trans::expr::trans_into::h77ab0e7120a822e9znh
  20:     0x7faa4e283817 - trans::controlflow::trans_block::h42681ce0640e2e4cb5d
  21:     0x7faa4e35ae11 - trans::base::trans_closure::hb2757f968ffe5e4fKkt
  22:     0x7faa4e26c0f8 - trans::base::trans_fn::hce2b3d6c72d05174Dvt
  23:     0x7faa4e26db90 - trans::monomorphize::monomorphic_fn::h49a5d55eb30e66a7usd
  24:     0x7faa4e2b343e - trans::callee::trans_fn_ref_with_substs::h3aa1af6a03cfb4944kg
  25:     0x7faa4e2b1a2e - trans::callee::trans_fn_ref::h8889dc3d233c59b9E9f
  26:     0x7faa4e2aee7d - trans::callee::trans::h4cf60639a99f29d7VYf
  27:     0x7faa4e2c55ab - trans::callee::trans_call_inner::h8632436444412771428
  28:     0x7faa4e2d1fd0 - trans::expr::trans_rvalue_dps_unadjusted::hba1263ec7559b76b4zi
  29:     0x7faa4e2cfc58 - trans::expr::trans_unadjusted::h99b640dbc3358c2cz4h
  30:     0x7faa4e284148 - trans::expr::trans::h5b717c9ef7c9e01eHth
  31:     0x7faa4e2c0fe3 - trans::callee::trans_args::h4b19cef3e74053e2m1g
  32:     0x7faa4e2c6490 - trans::callee::trans_call_inner::h8632436444412771428
  33:     0x7faa4e2d1fd0 - trans::expr::trans_rvalue_dps_unadjusted::hba1263ec7559b76b4zi
  34:     0x7faa4e2829e6 - trans::expr::trans_into::h77ab0e7120a822e9znh
  35:     0x7faa4e392f57 - trans::_match::mk_binding_alloca::h7291972955133085902
  36:     0x7faa4e281ded - trans::base::init_local::h638f4027bba7a7d4czs
  37:     0x7faa4e2834f2 - trans::controlflow::trans_block::h42681ce0640e2e4cb5d
  38:     0x7faa4e35ae11 - trans::base::trans_closure::hb2757f968ffe5e4fKkt
  39:     0x7faa4e26c0f8 - trans::base::trans_fn::hce2b3d6c72d05174Dvt
  40:     0x7faa4e267e91 - trans::base::trans_item::h61ea9df2ff905370vTt
  41:     0x7faa4e363ddc - trans::base::trans_crate::h1737e7beedd46e53GPu
  42:     0x7faa4f268b83 - driver::phase_4_translate_to_llvm::h3290bee8a52fe776rNa
  43:     0x7faa4f24453f - driver::compile_input::hb18fe8ea427236ffIba
  44:     0x7faa4f30c917 - run_compiler::h845d3feaee911021H5b
  45:     0x7faa4f30a489 - thunk::F.Invoke<A, R>::invoke::h16057606092771618870
  46:     0x7faa4f309100 - rt::unwind::try::try_fn::h9181260052925924808
  47:     0x7faa4eccc2f8 - rust_try_inner
  48:     0x7faa4eccc2e5 - rust_try
  49:     0x7faa4f30989f - thunk::F.Invoke<A, R>::invoke::h1492325410489626830
  50:     0x7faa4ec47665 - sys::thread::thread_start::h6486a3ef3bc76b1ar8E
  51:     0x7faa48d38181 - start_thread
  52:     0x7faa4e80047c - __clone
  53:                0x0 - <unknown>

**Comment by cbcrouch** [comment]
Source code can be found at:

https://github.com/cbcrouch/lang_exp/tree/master/rust/rust_app

Compiler info --version --verbose:

rustc 1.0.0-dev (3b3bb0e68 2015-03-04) (built 2015-03-04)
binary: rustc
commit-hash: 3b3bb0e682c2d252e9f62dd9df5cff9552af91ad
commit-date: 2015-03-04
build-date: 2015-03-04
host: x86_64-unknown-linux-gnu
release: 1.0.0-dev

**Comment by michaelwoerister** [comment]
Thanks for the detailed report!

**Comment by michaelwoerister** [comment]
This is very likely a duplicate of https://github.com/rust-lang/rust/issues/22432. I'm leaving it open for now.

**Comment by Ogeon** [comment]
A similar or the same ICE [popped up in Rustful](https://travis-ci.org/Ogeon/rustful/builds/53182371) with today's nightly (`rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-05)`). The difference I can see is that it's reported for `rwlock.rs`.

```
stack backtrace:
   1:     0x7f9dbf019f1f - sys::backtrace::write::hf79a3da4fdecb8a0OBA
   2:     0x7f9dbf044c32 - panicking::on_panic::h9f64f4c69e19f194hHJ
   3:     0x7f9dbef79eda - rt::unwind::begin_unwind_inner::h37f4496c980fe936knJ
   4:     0x7f9dbc3aaabd - rt::unwind::begin_unwind::h8320268356453106285
   5:     0x7f9dbc3aaa63 - diagnostic::SpanHandler::span_bug::h83c8af232eaba6a9h0D
   6:     0x7f9dbcc86ca3 - session::Session::span_bug::h857b2c7ae23c9286ISp
   7:     0x7f9dbe7a480c - trans::debuginfo::scope_metadata::hac54dfdbdcd04cd9SjE
   8:     0x7f9dbe6b7408 - trans::debuginfo::set_source_location::h1067a74086ed9dd48MD
   9:     0x7f9dbe66be42 - trans::expr::trans_into::h95c6d2681fdd2548znh
  10:     0x7f9dbe66c109 - trans::expr::trans_into::h95c6d2681fdd2548znh
  11:     0x7f9dbe6ce67f - trans::expr::trans_uniq_expr::h5f082eea62818f84ukj
  12:     0x7f9dbe6cf2ef - trans::expr::trans_unary::h4412379888608420Jgj
  13:     0x7f9dbe6b980e - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
  14:     0x7f9dbe66c417 - trans::expr::trans_into::h95c6d2681fdd2548znh
  15:     0x7f9dbe6ec6f9 - trans::expr::trans_adt::h1af69b9b4e52152aO6i
  16:     0x7f9dbe6ef1af - trans::expr::trans_struct::closure.42069
  17:     0x7f9dbe6d9016 - trans::expr::trans_struct::hcae8f9103f3460d5K2i
  18:     0x7f9dbe6bb53d - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  19:     0x7f9dbe66c3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  20:     0x7f9dbe66d227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
  21:     0x7f9dbe744821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
  22:     0x7f9dbe655b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
  23:     0x7f9dbe6575a0 - trans::monomorphize::monomorphic_fn::hc1b7393dd1dc77f1usd
  24:     0x7f9dbe69ce4e - trans::callee::trans_fn_ref_with_substs::hd01acb4398310d154kg
  25:     0x7f9dbe69b43e - trans::callee::trans_fn_ref::hb48e614c9b6dd9bcE9f
  26:     0x7f9dbe69888d - trans::callee::trans::ha56f4fe94448e6baVYf
  27:     0x7f9dbe6aefbb - trans::callee::trans_call_inner::h9722042290657949952
  28:     0x7f9dbe6bb9e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  29:     0x7f9dbe66c3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  30:     0x7f9dbe6ec6f9 - trans::expr::trans_adt::h1af69b9b4e52152aO6i
  31:     0x7f9dbe6ef1af - trans::expr::trans_struct::closure.42069
  32:     0x7f9dbe6d9016 - trans::expr::trans_struct::hcae8f9103f3460d5K2i
  33:     0x7f9dbe6bb53d - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
  34:     0x7f9dbe66c3f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
  35:     0x7f9dbe66d227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
  36:     0x7f9dbe744821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
  37:     0x7f9dbe655b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
  38:     0x7f9dbe652241 - trans::base::trans_item::h48fc370b7d259ac7vTt
  39:     0x7f9dbe652078 - trans::base::trans_item::h48fc370b7d259ac7vTt
  40:     0x7f9dbe74d7ec - trans::base::trans_crate::hc92be67ede893c70GPu
  41:     0x7f9dbf652e83 - driver::phase_4_translate_to_llvm::h9904f5d5fc3fb761rNa
  42:     0x7f9dbf62e83f - driver::compile_input::h3913ff7013f0c056Iba
  43:     0x7f9dbf6f6cb7 - run_compiler::h28a4446bae1034e7H5b
  44:     0x7f9dbf6f4829 - thunk::F.Invoke<A, R>::invoke::h6503055919709693733
  45:     0x7f9dbf6f34a0 - rt::unwind::try::try_fn::h1384674024000742916
  46:     0x7f9dbf0b4de8 - rust_try_inner
  47:     0x7f9dbf0b4dd5 - rust_try
  48:     0x7f9dbf6f3c3f - thunk::F.Invoke<A, R>::invoke::h5780663349966142752
  49:     0x7f9dbf02f965 - sys::thread::thread_start::h4ab695857833a5dar8E
  50:     0x7f9db8e8a181 - start_thread
  51:     0x7f9dbebea47c - __clone
  52:                0x0 - <unknown>
```

**Comment by kinghajj** [comment]
Another example, in kinghajj/deque@defb7983b4891c9a01168ad775a916c45c97479a,

    $ RUST_BACKTRACE=1 cargo test
       Compiling deque v0.1.8 (file:///home/samuel/deque)
    /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sync/mutex.rs:177:37: 180:2 error: internal compiler error: debuginfo: Could not find scope info for node NodeExpr(Expr { id: 3082, node: ExprStruct(Path { span: Span { lo: BytePos(4557025), hi: BytePos(4557036), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: StaticMutex#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }, [Field { ident: Spanned { node: lock#0, span: Span { lo: BytePos(1770102), hi: BytePos(1770106), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 3083, node: ExprPath(None, Path { span: Span { lo: BytePos(4557049), hi: BytePos(4557064), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: sys#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: MUTEX_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4557049), hi: BytePos(4557064), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4557043), hi: BytePos(4557064), expn_id: ExpnId(4294967295) } }, Field { ident: Spanned { node: poison#0, span: Span { lo: BytePos(1770129), hi: BytePos(1770135), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 3084, node: ExprPath(None, Path { span: Span { lo: BytePos(4557078), hi: BytePos(4557095), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: poison#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: FLAG_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4557078), hi: BytePos(4557095), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4557070), hi: BytePos(4557095), expn_id: ExpnId(4294967295) } }], None), span: Span { lo: BytePos(4557025), hi: BytePos(4557098), expn_id: ExpnId(4294967295) } })
    note: the compiler unexpectedly panicked. this is a bug.
    note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
    note: run with `RUST_BACKTRACE=1` for a backtrace
    thread 'rustc' panicked at 'Box<Any>', /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libsyntax/diagnostic.rs:129

    stack backtrace:
       1:     0x7f10c977cf1f - sys::backtrace::write::hf79a3da4fdecb8a0OBA
       2:     0x7f10c97a7c32 - panicking::on_panic::h9f64f4c69e19f194hHJ
       3:     0x7f10c96dceda - rt::unwind::begin_unwind_inner::h37f4496c980fe936knJ
       4:     0x7f10c6b10abd - rt::unwind::begin_unwind::h8320268356453106285
       5:     0x7f10c6b10a63 - diagnostic::SpanHandler::span_bug::h83c8af232eaba6a9h0D
       6:     0x7f10c73eeca3 - session::Session::span_bug::h857b2c7ae23c9286ISp
       7:     0x7f10c8f2180c - trans::debuginfo::scope_metadata::hac54dfdbdcd04cd9SjE
       8:     0x7f10c8e34408 - trans::debuginfo::set_source_location::h1067a74086ed9dd48MD
       9:     0x7f10c8de8e42 - trans::expr::trans_into::h95c6d2681fdd2548znh
      10:     0x7f10c8de9109 - trans::expr::trans_into::h95c6d2681fdd2548znh
      11:     0x7f10c8e4b67f - trans::expr::trans_uniq_expr::h5f082eea62818f84ukj
      12:     0x7f10c8e4c2ef - trans::expr::trans_unary::h4412379888608420Jgj
      13:     0x7f10c8e3680e - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
      14:     0x7f10c8de9417 - trans::expr::trans_into::h95c6d2681fdd2548znh
      15:     0x7f10c8e696f9 - trans::expr::trans_adt::h1af69b9b4e52152aO6i
      16:     0x7f10c8e6c1af - trans::expr::trans_struct::closure.42069
      17:     0x7f10c8e56016 - trans::expr::trans_struct::hcae8f9103f3460d5K2i
      18:     0x7f10c8e3853d - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
      19:     0x7f10c8de93f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
      20:     0x7f10c8dea227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
      21:     0x7f10c8ec1821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
      22:     0x7f10c8dd2b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
      23:     0x7f10c8dd45a0 - trans::monomorphize::monomorphic_fn::hc1b7393dd1dc77f1usd
      24:     0x7f10c8e19e4e - trans::callee::trans_fn_ref_with_substs::hd01acb4398310d154kg
      25:     0x7f10c8e1843e - trans::callee::trans_fn_ref::hb48e614c9b6dd9bcE9f
      26:     0x7f10c8e1588d - trans::callee::trans::ha56f4fe94448e6baVYf
      27:     0x7f10c8e2bfbb - trans::callee::trans_call_inner::h9722042290657949952
      28:     0x7f10c8e389e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
      29:     0x7f10c8e36668 - trans::expr::trans_unadjusted::hfd3a5e1b5cbe37d5z4h
      30:     0x7f10c8deab58 - trans::expr::trans::h23d7d0dd91a5190fHth
      31:     0x7f10c8e279f3 - trans::callee::trans_args::h29a92a6ed71c85ebm1g
      32:     0x7f10c8e2cea0 - trans::callee::trans_call_inner::h9722042290657949952
      33:     0x7f10c8e389e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
      34:     0x7f10c8de93f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
      35:     0x7f10c8e696f9 - trans::expr::trans_adt::h1af69b9b4e52152aO6i
      36:     0x7f10c8e6c1af - trans::expr::trans_struct::closure.42069
      37:     0x7f10c8e56016 - trans::expr::trans_struct::hcae8f9103f3460d5K2i
      38:     0x7f10c8e3853d - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
      39:     0x7f10c8de93f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
      40:     0x7f10c8dea227 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
      41:     0x7f10c8ec1821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
      42:     0x7f10c8dd2b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
      43:     0x7f10c8dd45a0 - trans::monomorphize::monomorphic_fn::hc1b7393dd1dc77f1usd
      44:     0x7f10c8e19e4e - trans::callee::trans_fn_ref_with_substs::hd01acb4398310d154kg
      45:     0x7f10c8e1843e - trans::callee::trans_fn_ref::hb48e614c9b6dd9bcE9f
      46:     0x7f10c8e1588d - trans::callee::trans::ha56f4fe94448e6baVYf
      47:     0x7f10c8e2bfbb - trans::callee::trans_call_inner::h9722042290657949952
      48:     0x7f10c8e389e0 - trans::expr::trans_rvalue_dps_unadjusted::hd47de7ac66e018254zi
      49:     0x7f10c8de93f6 - trans::expr::trans_into::h95c6d2681fdd2548znh
      50:     0x7f10c8ef9967 - trans::_match::mk_binding_alloca::h13035368140960659810
      51:     0x7f10c8de87fd - trans::base::init_local::h1e7c96bb7077440dczs
      52:     0x7f10c8de9f02 - trans::controlflow::trans_block::h3e86dfa8c58560e6b5d
      53:     0x7f10c8ec1821 - trans::base::trans_closure::hab3cc3c679d5ff23Kkt
      54:     0x7f10c8dd2b08 - trans::base::trans_fn::he0569b8eb832adf9Dvt
      55:     0x7f10c8dce8a1 - trans::base::trans_item::h48fc370b7d259ac7vTt
      56:     0x7f10c8eca7ec - trans::base::trans_crate::hc92be67ede893c70GPu
      57:     0x7f10c9db7e83 - driver::phase_4_translate_to_llvm::h9904f5d5fc3fb761rNa
      58:     0x7f10c9d9383f - driver::compile_input::h3913ff7013f0c056Iba
      59:     0x7f10c9e5bcb7 - run_compiler::h28a4446bae1034e7H5b
      60:     0x7f10c9e59829 - thunk::F.Invoke<A, R>::invoke::h6503055919709693733
      61:     0x7f10c9e584a0 - rt::unwind::try::try_fn::h1384674024000742916
      62:     0x7f10c9817de8 - rust_try_inner
      63:     0x7f10c9817dd5 - rust_try
      64:     0x7f10c9e58c3f - thunk::F.Invoke<A, R>::invoke::h5780663349966142752
      65:     0x7f10c9792965 - sys::thread::thread_start::h4ab695857833a5dar8E
      66:     0x7f10c35d7373 - start_thread
      67:     0x7f10c935827c - __clone
      68:                0x0 - <unknown>

**Comment by mikedilger** [comment]
I hit this trying to compile glutin.
rustc 1.0.0-nightly (3b3bb0e68 2015-03-04) (built 2015-03-05)

> /home/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-linux/build/src/libstd/sync/mutex.rs:177:37: 180:2 error: internal compiler error: debuginfo: Could not find scope info for node NodeExpr(Expr { id: 30848, node: ExprStruct(Path { span: Span { lo: BytePos(4847499), hi: BytePos(4847510), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: StaticMutex#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }, [Field { ident: Spanned { node: lock#0, span: Span { lo: BytePos(1770102), hi: BytePos(1770106), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 30849, node: ExprPath(None, Path { span: Span { lo: BytePos(4847523), hi: BytePos(4847538), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: sys#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: MUTEX_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4847523), hi: BytePos(4847538), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4847517), hi: BytePos(4847538), expn_id: ExpnId(4294967295) } }, Field { ident: Spanned { node: poison#0, span: Span { lo: BytePos(1770129), hi: BytePos(1770135), expn_id: ExpnId(4294967295) } }, expr: Expr { id: 30850, node: ExprPath(None, Path { span: Span { lo: BytePos(4847552), hi: BytePos(4847569), expn_id: ExpnId(4294967295) }, global: false, segments: [PathSegment { identifier: poison#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }, PathSegment { identifier: FLAG_INIT#0, parameters: AngleBracketedParameters(AngleBracketedParameterData { lifetimes: [], types: [], bindings: [] }) }] }), span: Span { lo: BytePos(4847552), hi: BytePos(4847569), expn_id: ExpnId(4294967295) } }, span: Span { lo: BytePos(4847544), hi: BytePos(4847569), expn_id: ExpnId(4294967295) } }], None), span: Span { lo: BytePos(4847499), hi: BytePos(4847572), expn_id: ExpnId(4294967295) } })

**Comment by cbcrouch** [comment]
This single line of code will cause the error:

let five_mut = Arc::new(Mutex::new(5));


rustc 1.0.0-dev (1fe8f2214 2015-03-06) (built 2015-03-06)
binary: rustc
commit-hash: 1fe8f221450bad3ffb1351c6549f67c18ce0b94e
commit-date: 2015-03-06
build-date: 2015-03-06
host: x86_64-unknown-linux-gnu
release: 1.0.0-dev

**Comment by michaelwoerister** [comment]
:police_car: Woh! Here comes the dupe-police! :police_car: :police_car:
This is a duplicate of #22432, a fix is already in the queue: #23074
Thanks for reporting! Keep your eyes open, citizens!

**Comment by Ogeon** [comment]
Thank you, officer!
